src/PSR7/Validators/Headers.php

```diff
@@ -20,22 +19,14 @@ class Headers
      */
     public function validate(MessageInterface $message, array $headerSpecs) : void
     {
-        $messageHeaders = $message->getHeaders();
+        // By default this will not report unexpected headers (soft validation)
+        // TODO, maybe this can be enabled later and controlled by custom options
+        // in such case throw new \RuntimeException($header, 200);
 
-        foreach ($messageHeaders as $header => $headerValues) {
-            if (! array_key_exists($header, $headerSpecs)) {
-                // By default this will not report unexpected headers (soft validation)
-                // TODO, maybe this can be enabled later and controlled by custom options
-                // throw new \RuntimeException($header, 200);
-                continue;
-            }
-
-            $validator = new SchemaValidator($this->detectValidationStrategy($message));
-            foreach ($headerValues as $headerValue) {
-                $validator->validate($headerValue, $headerSpecs[$header]->schema);
-            }
+        if (count($headerSpecs) == 0) {
```

Comment:
Here is no need in explicit check, since with empty count there will be just an empty foreach.